### DISCUSSION TITLE
aria-current and server components

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#49214] [Help] [answered]
### Summary

i can render a `next/link` as a server component, however when i add `aria-current` to the link it seems i need to switch it to a client component - is there a way to avoid this? (note: it's totally fine to render as a client component, i'm just asking for my understanding)

```ts
"use client";

import Link, { type LinkProps } from "next/link";
import { usePathname } from "next/navigation";
import { type ReactNode } from "react";

interface NavLinkProps) {
  children: ReactNode;
  href: Exclude<LinkProps["href"], string>;
}

export function NavLink(props: NavLinkProps) {
  const { children, href } = props;
  const pathname = usePathname();
  const isCurrentPage = href.pathname === pathname;

  return <Link aria-current={isCurrentPage ? 'page' : undefined} {...props} />
}
```

**Comment by oliverjam** [comment, marked as answer]
The `usePathname` hook can [only be used in Client Components](https://nextjs.org/docs/app/api-reference/functions/use-pathname#good-to-know). I assume this is because it relies on React Context to know which route it was rendered within (and Context doesn't work in Server Components).

As far as I can tell [there's no easy way to get the URL](https://github.com/vercel/next.js/issues/47108) of the current request for a Server Component. They only receive `params` and `searchParams` as props, and can only retrieve the headers and cookies using built-in Next functions. They cannot access the underlying HTTP request or URL.

Hopefully Next will add a built-in way to access the HTTP request, otherwise it'll be impossible to achieve basic features like this server-side 🤷
